test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java

```diff
@@ -101,6 +102,24 @@ protected Settings nodeSettings(int nodeOrdinal) {
         return Arrays.asList(MockRepository.Plugin.class);
     }
 
+    @After
```

Comment:
I like the idea of this, but I'm a little suspicious that this is the right place to put this. Many tests don't reuse the test cluster in which case the work here is unnecessary? Maybe we should make this change in the logic that cleans up the test cluster when its reused instead?

Reply:
The problem is that `@After` hooks from `AbstractSnapshotIntgetTestCase` are executed before the hooks in `ESIntegTestCase`. Is there a way to determine the scope of a cluster? we could bypass this cleanup if the cluster is not reused?

Reply:
Maybe we can do a simpler thing and simply skip the repo consistency checks in case of test failure via wrapping in:

```java
if (getSuiteFailureMarker().wasSuccessful()) {

}
```

That should fix all the cases because I think closing the repos upstream in the `EsIntegTestCase` cleanup logic should deal with unblocking and removing all the repos cleanly.
That would also make test failures easier to interpret since we get rid of failed repo verifications on failed tests?